artiq/coredevice/ad53xx.py

```diff
@@ -160,6 +160,10 @@ def __init__(self, dmgr, spi_device, ldac_device=None, clr_device=None,
         self.offset_dacs = offset_dacs
         self.core = dmgr.get(core)
 
+    @staticmethod
```

Comment:
Please just check whether the class has the method and default to the empty list if it doesn't. That's less invasive and easier to maintain than having to add the same trivial default implementation to all classes.

Reply:
We had a discussion with @sbourdeauducq , and agreed on explicitly failing if somebody forgot to implement this method.

Reply:
And given that already you in this first PR take all kinds of shortcuts (e.g. not filling in any channel names for phaser and elsewhere, foregoing possible crosschecks of enumeration consistency), I don't see why requiring an explicit default implementation everywhere helps.

Reply:
I never understood the reason for the self imposed limitation to generating the ddb template and rtiomap with stand alone tooling. Seems arbitrary. Just generate them at Gateware build time and either fuse them at load/flash time or (worse) using a trivial stand alone tool.

Reply:
But let's not shift the argument. This is originally about the need to copy paste function-less boilerplate into the core device drivers instead of defaulting.

Reply:
Each.

What's with Phaser? It doesn't speak DRTIO.

Reply:
> And there are many DRTIO systems which will have the problem anyway.

"The problem" being what now?